core/src/test/scala/unit/kafka/server/RequestUtilitiesTest.scala

```diff
@@ -0,0 +1,152 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package unit.kafka.server
+
+import kafka.network.SocketServer
+import kafka.server.{ControllerServer, IntegrationTestUtils, KafkaBroker}
+import kafka.test.ClusterInstance
+import kafka.utils.TestUtils
+import org.apache.kafka.clients.producer.{KafkaProducer, ProducerRecord, RecordMetadata}
+import org.apache.kafka.common.{TopicIdPartition, TopicPartition, Uuid}
+import org.apache.kafka.common.network.ListenerName
+import org.apache.kafka.common.requests.{AbstractRequest, AbstractResponse}
+import org.apache.kafka.common.security.auth.SecurityProtocol
+import org.apache.kafka.common.serialization.StringSerializer
+import org.apache.kafka.controller.ControllerRequestContextUtil.ANONYMOUS_CONTEXT
+import org.junit.jupiter.api.AfterEach
+
+import java.util.Properties
+import java.util.stream.Collectors
+import scala.collection.Seq
+import scala.collection.convert.ImplicitConversions.{`collection AsScalaIterable`, `map AsScala`}
+import scala.reflect.ClassTag
+
+class RequestUtilitiesTest(cluster: ClusterInstance) {
+
+  protected var producer: KafkaProducer[String, String] = _
+
+  protected def securityProtocol: SecurityProtocol = SecurityProtocol.PLAINTEXT
+  protected def listenerName: ListenerName = ListenerName.forSecurityProtocol(securityProtocol)
+
+  private def brokers(): Seq[KafkaBroker] = cluster.brokers.values().stream().collect(Collectors.toList[KafkaBroker]).toSeq
+
+  private def controllerServers(): Seq[ControllerServer] = cluster.controllers().values().toSeq
+
+  @AfterEach
+  def tearDown(): Unit = {
+    if (producer != null)
+      producer.close()
+  }
+
+  protected def createOffsetsTopic(): Unit = {
+    TestUtils.createOffsetsTopicWithAdmin(
+      admin = cluster.createAdminClient(),
+      brokers = brokers(),
+      controllers = controllerServers()
+    )
+  }
+
+  protected def createTopic(topic: String,
```

Comment:
This is good and potentially we should re-use this topic at other places in the test files. It will be good to create a jira atleast, outside kip-932, for the cleanup.